### DISCUSSION TITLE
rkyv

### DISCUSSION THREAD

**Discussion by ThomasCartier** [maxcountryman/tower-sessions#209] [Ideas]
Hi, 

I just used your lib with success. Thanks a lot.

2 ideas: 

1. Rkyv could help to reach more throughput by avoiding deserialization when not needed.
2. Instead of getting strings, could we accept to register enum values ? its would be faster.

Both could be enabled by features to satisfy everybody.

What's your opinion ?

**Comment by maxcountryman** [comment]
My opinion is this almost certainly doesn't matter in practice.

To prove that it does would require:

1. Benchmarks to demonstrate a meaningful improvement that's likely to matter in the real world,
2. And a real world use case that demonstrates this is a meaningful bottleneck in a real workload.

Without these, I don't think it makes sense to pursue.
